jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/HTTP3StreamConnection.java

```diff
@@ -125,9 +125,11 @@ private void processDataFrames(boolean setFillInterest)
                 case FRAME ->
                 {
                     action.getAndSet(null).run();
-                    // Release the network buffer here (if empty), since the application may
-                    // not be reading more bytes, to avoid to keep around a consumed buffer.
-                    tryReleaseBuffer(false);
+                    // We cannot release the buffer eagerly here as that would race
+                    // with another thread executing user code calling Stream.read();
+                    // that user thread could throw NPE in HTTP3StreamConnection.parseAndFill()
+                    // when trying to access the networkBuffer field that is set
+                    // to null here depending on which thread wins the race - see #9765.
```

Comment:
I'd simplify to:

```suggestion
                    // Do not release the buffer here to avoid races with 
                    // user-spawned threads that may call Stream.read().
```

The reason to say less is to avoid that the comment rots if we change the implementation.

Reply:
Done.